run/hello-broken/main.py

```diff
@@ -0,0 +1,71 @@
+# Copyright 2019 Google, LLC.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from flask import Flask
+import json
+import os
+import sys
+
+app = Flask(__name__)
+
+# [START run_broken_service]
+@app.route("/", methods=["GET"])
+def index():
+    print("hello: received request.")
+
+    # [START run_broken_service_problem]
+    NAME = os.getenv("NAME")
+
+    if not NAME:
+        print("Environment validation failed.")
+        raise Exception("Missing required service parameter.")
+    # [END run_broken_service_problem]
+
+    # Flush the stdout to avoid log buffering.
+    sys.stdout.flush()
+
+    return f"Hello {NAME}"
+# [END run_broken_service]
+
+
+@app.route("/improved", methods=["GET"])
+def improved():
+    print("hello: received request.")
+
+    # [START run_broken_service_upgrade]
```

Comment:
should the region tag be in here or should it be around the whole function? and does it need imports?

Reply:
Should only be around the "fixed" part of the previously broken function!  [Node.js example](https://github.com/GoogleCloudPlatform/nodejs-docs-samples/blob/master/run/hello-broken/index.js#L38).